### ISSUE TITLE
AWSCloudWatchProvider fails to log to a brand-new log stream

### ISSUE THREAD

**Issue by ibliskavka** [aws-amplify/amplify-js#9050]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

Not applicable

### Amplify Categories

Not applicable

### Environment information

<details>

# Put output below this line

  System:
    OS: Linux 5.10 Ubuntu 20.04.3 LTS (Focal Fossa)
    CPU: (12) x64 Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz
    Memory: 9.66 GB / 12.33 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 14.17.5 - /usr/bin/node
    Yarn: 1.22.11 - /mnt/c/Users/iblis/AppData/Roaming/npm/yarn
    npm: 6.14.14 - /usr/bin/npm
  npmGlobalPackages:
    aws-cdk: 1.122.0
    npm: 6.14.14

</details>

### Describe the bug

AWSCloudWatchProvider (`packages/core/src/Providers/AWSCloudWatchProvider.ts`) throws the following errors when sending logs to a brand new stream.

**API Response**
```
{"__type":"InvalidParameterException","message":"1 validation error detected: Value '' at 'sequenceToken' failed to satisfy constraint: Member must have length greater than or equal to 1"}
```

When I call the CloudWatch API directly without a token to initialize the first stream message, the provider works.

### Expected behavior

`AWSCloudWatchProvider` shouldn't be initializing `_nextSequenceToken` to empty string. The value should be undefined.

**Original Line: 379**
`this._nextSequenceToken = logStream.uploadSequenceToken || '';`

**Requested Changes:**
`this._nextSequenceToken = logStream.uploadSequenceToken || undefined;`

### Reproduction steps

1. Create a new aws-amplify logger
2. Add Pluggable: `AWSCloudWatchProvider` with a non-existing stream name
3. Attempt to log an entry

### Code Snippet

```javascript
const logger = new Logger('test', 'DEBUG);
logger.addPluggable(
    new AWSCloudWatchProvider({
      logGroupName: `/app/dev`,
      logStreamName: 'brand-new-stream',
      region: 'us-east-2`,
    })
  );

logger.debug('hello');
```

### Log output

<details>
**API Response**
```
{"__type":"InvalidParameterException","message":"1 validation error detected: Value '' at 'sequenceToken' failed to satisfy constraint: Member must have length greater than or equal to 1"}
```

**Log Output**
```
31:07.130 AWSCloudWatch - failure during log push: InvalidParameterException: 1 validation error detected: Value '' at 'sequenceToken' failed to satisfy constraint: Member must have length greater than or equal to 1

31:07.130 AWSCloudWatch - error during _safeUploadLogEvents: TypeError: Cannot read properties of undefined (reading 'nextSequenceToken')

[ERROR] 31:07.130 AWSCloudWatch - error when calling _safeUploadLogEvents in the timer interval - TypeError: Cannot read properties of undefined (reading 'nextSequenceToken')
```
</details>

**Comment by chrisbonifacio** [comment]
Hi 👋 Closing this as we have not heard back from you. If you are still experiencing this issue and in need of assistance, please feel free to comment and provide us with any information previously requested by our team members so we can re-open this issue and be better able to assist you. Thank you!

**Comment by aaronlna** [comment]
Why was this issue closed? This issue still exists, I ran into it this morning. The steps that were originally posted look correct in how to reproduce the problem.

When writing to an empty log-stream for the first time then all writes will fail because `_nextSequenceToken` will be an empty string that fails input validation to `PutLogEventsCommand`. This means you can never begin writing events. I've had to work around this by publishing an event manually, but this isn't feasible for my product that will need to generate new log streams per different sources. We won't have a mechanism in this case for writing an event out-of-band to a new, empty log stream.

I'm happy to post a PR.

**Comment by slavco86** [comment]
Same here! When trying to follow instructions in #8309 to get Amplify to send logs to Cloudwatch, at first I was unable to because of exactly same error - sequence token.
Once I have manually added a log inside the stream with some dummy data - everything worked as expected

**Comment by ashika01** [comment]
I have merged in #9181. Thank you @aaronlna. It is now in `aws-amplify@unstable` and should be released in `@latest` thursday during our weekly release cycle. Could you verify by upgrading to `unstable` and let us know if there are any issues?

**Comment by aaronlna** [comment]
Thanks for the merge. I checked out `unstable` and it is working for my use-case-- no issues discovered.
